src/system.rs

```diff
@@ -1,14 +1,25 @@
+use alloc::boxed::Box;
+use alloc::rc::Rc;
+use alloc::string::String;
+use alloc::vec::Vec;
+use core::cell::RefCell;
+
+use anyhow::anyhow;
+
+use crankstart_sys::ctypes::{c_char, c_int};
+pub use crankstart_sys::PDButtons;
+use crankstart_sys::PDMenuItem;
 use {
-    crate::pd_func_caller, alloc::format, anyhow::Error, core::ptr, crankstart_sys::ctypes::c_void,
+    crate::pd_func_caller, anyhow::Error, core::ptr, crankstart_sys::ctypes::c_void,
     cstr_core::CString,
 };
 
-use crankstart_sys::ctypes::c_int;
-pub use crankstart_sys::PDButtons;
-use crankstart_sys::{PDDateTime, PDLanguage, PDPeripherals};
-
 static mut SYSTEM: System = System(ptr::null_mut());
 
+static mut MENU_ITEM_CALLBACKS: Vec<Box<dyn Fn()>> = Vec::new();
```

Comment:
Not sure I understand how it used. Could you explain please?

Reply:
Oops sorry, poor review on my part these are left over from original implementations. I've removed them now in this PR